python/lisp.py

```diff
@@ -1,76 +1,79 @@
 import sys
-from enum import Enum
-from typing import Any, Tuple
+from enum import auto, Enum, unique
+from typing import Any, NamedTuple, Optional, Tuple
 
 
+@unique
 class TokenKind(Enum):
-    INTEGER = 1
-    IDENTIFIER = 2
-    SYNTAX = 3
+    INTEGER = auto()
+    IDENTIFIER = auto()
+    SYNTAX = auto()
 
-    
-class Token():
+
+class Token(NamedTuple):
     value: str
     kind: TokenKind
 
-    def __init__(self, value, kind):
-        self.value = value
-        self.kind = kind
-
 
+@unique
 class SexpKind(Enum):
-    ATOM = 1
-    PAIR = 2
+    ATOM = auto()
+    PAIR = auto()
 
 
-class Sexp():
+class Sexp(NamedTuple):
```

Comment:
```py
from dataclasses import dataclass
@dataclass
class Sexp:
    ...
```

Reply:
Yes, also good. Although in this case, I went for `NamedTuple` because it saves a line per usage. :)